reference/glossary.md

```diff
@@ -0,0 +1,17 @@
+(reference-glossary)=
+# Glossary
+
+```{glossary}
+
+Amazon Web Services (AWS)
+    A cloud platform provided by Amazon that can be used to host Charmed HPC. See the [AWS website](https://aws.amazon.com/) for more information.
+
+Graphics Processing Unit (GPU)
+    A specialized processor that is designed to accelerate image processing and graphics rendering for output to a display device.
```

Comment:
I'm not sure this is the best definition to use for GPUs. My reasoning is that the definition doesn't mention GPUs use case in HPC, but rather what they were traditionally used for in personal computers.

We should rework this definition a bit to be more relevant to how GPUs are used within the context of Charmed HPC. E.g. how they are used for AI/ML workloads, matrix calculations, etc.